src/main/java/org/codehaus/plexus/util/xml/Xpp3DomBuilder.java

```diff
@@ -194,4 +217,12 @@ else if ( eventType == XmlPullParser.END_TAG )
 
         throw new IllegalStateException( "End of document found before returning to 0 depth" );
     }
+
+    /**
+     * Input location builder interface, to be implemented to choose how to store data.
+     */
+    public static interface InputLocationBuilder
+    {
+        Object toInputLocation( XmlPullParser parser );
```

Comment:
What about 'getCurrentInputLocation'?

Should we explicitly state the expected behavior?

Is it better to return an GenericInputLocation, just as a placeholder interface? It will make code more typesafe and clearer IMHO

Reply:
'getCurrentInputLocation': since it is expected to create an object, I don't like 'get'

What do you mean by "state the expected behavior"?

Since the implementation is really something that will be specific to the user, and that the user will have to cast, I don't see the benefit of creating an empty interface

Reply:
> 'getCurrentInputLocation': since it is expected to create an object, I don't like 'get'
> 
My concern is that the function takes a stateful object as parameter, so I image it will return a view of the current state of the parser.
the word 'to' seems to me like a 'conversion'.
Not so important to me, I am very new to this code.


> What do you mean by "state the expected behavior"?
> 
For instance, if the method is called twice is it expected to return two *equivalent* objects, the same object, or what ?

```
Object a =  toInputLocation( parser );
Object b =  toInputLocation( parser );
```
should b == a, b.quals(a) or it is not important ?



> Since the implementation is really something that will be specific to the user, and that the user will have to cast, I don't see the benefit of creating an empty interface

Fine

Reply:
for input location, ordering is not important: really, the documentation to read is https://codehaus-plexus.github.io/modello/location-tracking.html